diff --git a/conda/pytorch-sparse/build_conda.sh b/conda/pytorch-sparse/build_conda.sh
index ee8e780..d0491e3 100755
--- a/conda/pytorch-sparse/build_conda.sh
+++ b/conda/pytorch-sparse/build_conda.sh
@@ -4,7 +4,7 @@ export PYTHON_VERSION=$1
 export TORCH_VERSION=$2
 export CUDA_VERSION=$3
 
-export CONDA_PYTORCH_CONSTRAINT="pytorch==${TORCH_VERSION%.*}.*"
+export CONDA_PYTORCH_CONSTRAINT="pytorch==${TORCH_VERSION} *_openmpi"
 
 if [ "${CUDA_VERSION}" = "cpu" ]; then
   export CONDA_CUDATOOLKIT_CONSTRAINT="cpuonly  # [not osx]"
@@ -30,4 +30,4 @@ echo "PyTorch $TORCH_VERSION+$CUDA_VERSION"
 echo "- $CONDA_PYTORCH_CONSTRAINT"
 echo "- $CONDA_CUDATOOLKIT_CONSTRAINT"
 
-conda build . -c pytorch -c pyg -c default -c nvidia --output-folder "$HOME/conda-bld"
+conda build . --override-channels -c "$HOME/conda-bld" -c katanagraph -c pytorch -c pyg -c conda-forge -c nvidia --output-folder "$HOME/conda-bld"
diff --git a/conda/pytorch-sparse/meta.yaml b/conda/pytorch-sparse/meta.yaml
index 6099738..98d36c6 100644
--- a/conda/pytorch-sparse/meta.yaml
+++ b/conda/pytorch-sparse/meta.yaml
@@ -23,11 +23,12 @@ requirements:
     - {{ environ.get('CONDA_CUDATOOLKIT_CONSTRAINT') }}
 
 build:
-  string: py{{ environ.get('PYTHON_VERSION').replace('.', '') }}_torch_{{ environ['TORCH_VERSION'] }}_{{ environ['CUDA_VERSION'] }}
+  number: 1
+  string: py{{ environ.get('PYTHON_VERSION').replace('.', '') }}_torch_{{ environ['TORCH_VERSION'] }}_{{ environ['CUDA_VERSION'] }}_openmpi
   script_env:
     - FORCE_CUDA
     - TORCH_CUDA_ARCH_LIST
-    - WITH_METIS=1
+    - WITH_METIS=0
   preserve_egg_dir: True
 
 test:
