diff --git a/conda/pytorch-spline-conv/build_conda.sh b/conda/pytorch-spline-conv/build_conda.sh
index 7dcdcab..11ba1a8 100755
--- a/conda/pytorch-spline-conv/build_conda.sh
+++ b/conda/pytorch-spline-conv/build_conda.sh
@@ -4,7 +4,7 @@ export PYTHON_VERSION=$1
 export TORCH_VERSION=$2
 export CUDA_VERSION=$3
 
-export CONDA_PYTORCH_CONSTRAINT="pytorch==${TORCH_VERSION%.*}.*"
+export CONDA_PYTORCH_CONSTRAINT="pytorch==${TORCH_VERSION} *_openmpi"
 
 if [ "${CUDA_VERSION}" = "cpu" ]; then
   export CONDA_CUDATOOLKIT_CONSTRAINT="cpuonly  # [not osx]"
@@ -40,7 +40,7 @@ echo "- $CONDA_PYTORCH_CONSTRAINT"
 echo "- $CONDA_CUDATOOLKIT_CONSTRAINT"
 
 if [ "${CUDA_VERSION}" = "cu116" ]; then
-  conda build . -c pytorch -c default -c nvidia -c conda-forge --output-folder "$HOME/conda-bld"
+  conda build . --override-channels -c "$HOME/conda-bld" -c katanagraph -c pytorch -c nvidia -c conda-forge --output-folder "$HOME/conda-bld"
 else
-  conda build . -c pytorch -c default -c nvidia --output-folder "$HOME/conda-bld"
+  conda build . --override-channels -c "$HOME/conda-bld" -c katanagraph -c pytorch -c nvidia --output-folder "$HOME/conda-bld"
 fi
diff --git a/conda/pytorch-spline-conv/meta.yaml b/conda/pytorch-spline-conv/meta.yaml
index 2c0a6e4..27550a5 100644
--- a/conda/pytorch-spline-conv/meta.yaml
+++ b/conda/pytorch-spline-conv/meta.yaml
@@ -21,7 +21,8 @@ requirements:
     - {{ environ.get('CONDA_CUDATOOLKIT_CONSTRAINT') }}
 
 build:
-  string: py{{ environ.get('PYTHON_VERSION').replace('.', '') }}_torch_{{ environ['TORCH_VERSION'] }}_{{ environ['CUDA_VERSION'] }}
+  number: 1
+  string: py{{ environ.get('PYTHON_VERSION').replace('.', '') }}_torch_{{ environ['TORCH_VERSION'] }}_{{ environ['CUDA_VERSION'] }}_openmpi
   script: pip install .
   script_env:
     - FORCE_CUDA
